tensorflow/lite/experimental/micro/examples/mnist_demo/main.cc

```diff
@@ -0,0 +1,140 @@
+#include "tensorflow/lite/experimental/micro/micro_error_reporter.h"
+#include "tensorflow/lite/experimental/micro/micro_interpreter.h"
+#include "tensorflow/lite/schema/schema_generated.h"
+#include "tensorflow/lite/version.h"
+
+// Helper fn to convert TFliteTypes into human readable string format
+const char* typeString(TfLiteType type) {
+  static const char noTypeStr[] = "NoType";
+  static const char float32Str[] = "Float32";
+  static const char int32Str[] = "Int32";
+  static const char uInt8Str[] = "UInt8";
+  static const char int64Str[] = "Int64";
+  static const char stringStr[] = "String";
+  static const char boolStr[] = "Bool";
+  static const char int16Str[] = "Int16";
+  static const char complex64Str[] = "Complex64";
+  static const char int8Str[] = "Int8";
+
+  switch (type) {
+    case kTfLiteFloat32:
+      return float32Str;
+    case kTfLiteInt32:
+      return int32Str;
+    case kTfLiteUInt8:
+      return uInt8Str;
+    case kTfLiteInt64:
+      return int64Str;
+    case kTfLiteString:
+      return stringStr;
+    case kTfLiteBool:
+      return boolStr;
+    case kTfLiteInt16:
+      return int16Str;
+    case kTfLiteComplex64:
+      return complex64Str;
+    case kTfLiteInt8:
+      return int8Str;
+    default:
+      return noTypeStr;
+  }
+}
+
+// Helper fn to log the shape and datatype of a tensor
+void printTensorDetails(TfLiteTensor* tensor,
+                        tflite::ErrorReporter* error_reporter) {
+  error_reporter->Report("Type [%s] Shape :", typeString(tensor->type));
+  for (int d = 0; d < tensor->dims->size; ++d) {
+    error_reporter->Report("%d [ %d]", d, tensor->dims->data[d]);
+  }
+  error_reporter->Report("");
+}
+
+int main(int argc, char* argv[]) {
+  // Set up logging.
+  tflite::MicroErrorReporter micro_error_reporter;
+  tflite::ErrorReporter* error_reporter = &micro_error_reporter;
+
+  // Map the model into a usable data structure. This doesn't involve any
+  // copying or parsing, it's a very lightweight operation.
+  error_reporter->Report("Parsing MNIST classifier model FlatBuffer.");
+  const tflite::Model* model = ::tflite::GetModel(mnist_dense_model_tflite);
+  if (model->version() != TFLITE_SCHEMA_VERSION) {
+    error_reporter->Report(
+        "Model provided is schema version %d not equal "
+        "to supported version %d.\n",
+        model->version(), TFLITE_SCHEMA_VERSION);
+    return 1;
+  }
+  error_reporter->Report("Model parsed.\n");
+
+  // This pulls in all the operation implementations we need.
+  tflite::ops::micro::AllOpsResolver resolver;
+
+  // Create an area of memory to use for input, output, and intermediate arrays.
+  // The size of this will depend on the model you're using, currently
+  // determined by experimentation.
+  const int tensor_arena_size = 200 * 1024;
+  uint8_t tensor_arena[tensor_arena_size];
+
+  // Build an interpreter to run the model with.
+  error_reporter->Report("Build interpreter.");
```

Comment:
We tend to avoid log statements scattered throughout the code.  While it's nice for debugging, we want only key relevant information to get logged, like output and errors.

Reply:
I've removed a few of the log statements so they're only shown for important steps and results now.